### DISCUSSION TITLE
useContract or useContractWrite

### DISCUSSION THREAD

**Discussion by Hussainzz** [wevm/wagmi#262] [Q&A] [answered]
so is it fine to use. useContract to interact with the deployed contract or is it better to use. useContractWrite/ useContractRead ?

**Comment by tmm** [comment, marked as answer]
I would recommend `useContractWrite` and `useContractRead`. They are allow declarative access to these methods instead of needs to manage state and errors yourself. In the next version of wagmi, there will also be some caching improvements for `useReadContract` as well as broader finite state changes.
